### PR TITLE
[FrameworkBundle] [HttpKernel] hotfix: fix test on master for php 7.1

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | 

This is fixing the test, but is it ok ? that's the question ;).

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Tests/Controller/ControllerResolverTest.php [hunk 1]
```diff
@@ -194,7 +196,7 @@ public function getUndefinedControllers()
         );
     }
 
-    protected function createControllerResolver(LoggerInterface $logger = null): ControllerResolverInterface
+    protected function createControllerResolver(LoggerInterface $logger = null, ContainerInterface $container = null, ControllerNameParser $parser = null): ControllerResolverInterface
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/Tests/Controller/ControllerResolverTest.php]: The HttpKernel does not know about ControllerNameParser. So it needs a different solution.

**Comment by Simperfit** [comment]
@Tobion updated.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerResolverTest.php [hunk 2]
```diff
@@ -169,13 +167,19 @@ class_exists(AbstractControllerTest::class);
         $this->assertSame($controllerContainer, $controller->getContainer());
     }
 
-    protected function createControllerResolver(LoggerInterface $logger = null, Psr11ContainerInterface $container = null, ControllerNameParser $parser = null): ControllerResolverInterface
+    protected function createControllerResolver(LoggerInterface $logger = null): ControllerResolverInterface
     {
-        if (!$parser) {
+        $parser = $container = null;
+
+        if (isset(\func_get_args()[2])) {
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerResolverTest.php]: we use `\func_num_args() >= 3` anywhere else instead

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerResolverTest.php [hunk 3]
```diff
@@ -169,13 +167,19 @@ class_exists(AbstractControllerTest::class);
         $this->assertSame($controllerContainer, $controller->getContainer());
     }
 
-    protected function createControllerResolver(LoggerInterface $logger = null, Psr11ContainerInterface $container = null, ControllerNameParser $parser = null): ControllerResolverInterface
+    protected function createControllerResolver(LoggerInterface $logger = null): ControllerResolverInterface
     {
-        if (!$parser) {
+        $parser = $container = null;
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerResolverTest.php]: this is not necessary, both variables are overwritten below anyway

File: src/Symfony/Component/HttpKernel/Tests/Controller/ContainerControllerResolverTest.php [hunk 4]
```diff
@@ -204,9 +204,12 @@ public function getUndefinedControllers()
         return $tests;
     }
 
-    protected function createControllerResolver(LoggerInterface $logger = null, ContainerInterface $container = null): ControllerResolverInterface
+    protected function createControllerResolver(LoggerInterface $logger = null): ControllerResolverInterface
     {
-        if (!$container) {
+        $container = null;
```
Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/Tests/Controller/ContainerControllerResolverTest.php]: not needed

**Comment by nicolas-grekas** [comment]
replaced by #26266
thanks you @Simperfit

**Pull request closed without merging**